app/scripts/detect-multiple-instances.test.js

```diff
@@ -0,0 +1,104 @@
+import { strict as assert } from 'assert';
+import browser from 'webextension-polyfill';
+import sinon from 'sinon';
+import {
+  PLATFORM_CHROME,
+  PLATFORM_EDGE,
+  METAMASK_BETA_CHROME_ID,
+  METAMASK_PROD_CHROME_ID,
+  METAMASK_FLASK_CHROME_ID,
+} from '../../shared/constants/app';
+import {
+  checkForMultipleVersionsRunning,
+  onMessageReceived,
+} from './detect-multiple-instances';
+import * as util from './lib/util';
+
+describe('multiple instances running detector', function () {
+  const PING_MESSAGE = 'isRunning';
+
+  let sendMessageStub = sinon.stub();
+
+  beforeEach(async function () {
+    sinon.replace(browser, 'runtime', {
+      sendMessage: sendMessageStub,
+      id: METAMASK_BETA_CHROME_ID,
+    });
+
+    sinon.stub(util, 'getPlatform').callsFake((_) => {
+      return PLATFORM_CHROME;
+    });
+  });
+
+  afterEach(function () {
+    sinon.restore();
+  });
+
+  describe('checkForMultipleVersionsRunning', function () {
+    it('should send ping message to multiple instances', async function () {
+      await checkForMultipleVersionsRunning();
+
+      assert(sendMessageStub.calledTwice);
+      assert(
+        sendMessageStub
+          .getCall(0)
+          .calledWithExactly(METAMASK_PROD_CHROME_ID, PING_MESSAGE),
+      );
+      assert(
+        sendMessageStub
+          .getCall(1)
+          .calledWithExactly(METAMASK_FLASK_CHROME_ID, PING_MESSAGE),
+      );
+    });
+
+    it('should not send ping message if platform is not Chrome or Firefox', async function () {
+      util.getPlatform.restore();
+      sendMessageStub = sinon.stub();
+
+      sinon.stub(util, 'getPlatform').callsFake((_) => {
+        return PLATFORM_EDGE;
+      });
+
+      await checkForMultipleVersionsRunning();
+
+      assert(sendMessageStub.notCalled);
+    });
+
+    it('should not expose an error outside if sendMessage throws', async function () {
+      sinon.restore();
+
+      sinon.replace(browser, 'runtime', {
+        sendMessage: sinon.stub().throws(),
+        id: METAMASK_BETA_CHROME_ID,
+      });
+
+      const spy = sinon.spy(checkForMultipleVersionsRunning);
+
+      await checkForMultipleVersionsRunning();
+
+      assert(!spy.threw());
```

Comment:
:sweat_smile: is this just to pass the lint rule about each test needing an assertion? This is a bit silly because this would have thrown on line 77. 

The better way to do this would probably be to use Jest's `expect(...).not.toThrow()` matcher. But I'm guessing this file is still running on mocha - it seems to be anyway. I should have suggested using Jest instead but I won't block on it.

Reply:
Yes, this one is running on mocha, so I couldn't use more suitable matcher 🙂 
The idea was to simulate throw on `sendMessage` and verify that it was handled inside and not exposed out, so this method is fail-safe